k9mail/src/main/java/com/fsck/k9/mailstore/LocalFolder.java

```diff
@@ -183,10 +188,23 @@ public Void doDbWork(final SQLiteDatabase db) throws WrappedException {
         }
     }
 
-    void open(Cursor cursor) throws MessagingException {
+    public void open(Cursor cursor) throws MessagingException {
         mFolderId = cursor.getInt(LocalStore.FOLDER_ID_INDEX);
         mName = cursor.getString(LocalStore.FOLDER_NAME_INDEX);
         mVisibleLimit = cursor.getInt(LocalStore.FOLDER_VISIBLE_LIMIT_INDEX);
+
+        //The index check here is because of an issue that occurs while upgrading the db to V55 - at that point, the
+        //uid_validity and highest_mod_Seq columns do not exist yet
```

Comment:
The problem is that the `open(Cursor)` method is called when upgrading to V55, but the `uid_validity` and `highest_mod_seq` columns do not exist yet. They are added only in V61. There does not seem to be a reliable way to recreate the open(Cursor) method in `MigrationTo61` .